apps/src/gamelab/GameLab.js

```diff
@@ -1208,6 +1202,38 @@ GameLab.prototype.onP5Preload = function () {
   return false;
 };
 
+GameLab.prototype.loadValidationCodeIfNeeded_ = function () {
+  if (this.level.validationCode && !this.level.helperLibraries.some(name => name === validationLibraryName)) {
+    this.level.helperLibraries.unshift(validationLibraryName);
+  }
+};
+
+let libraryPreload;
+GameLab.prototype.loadLibraries_ = function () {
+  if (!libraryPreload) {
+    this.loadValidationCodeIfNeeded_();
+    libraryPreload = Promise.all(this.level.helperLibraries.map(this.loadLibrary_.bind(this)));
+  }
+  return libraryPreload;
+};
+
+GameLab.prototype.loadLibrary_ = function (name) {
```

Comment:
This looks like a nice place for `async`:
```
GameLab.prototype.loadLibrary_ = async function (name) {
  if (this.libraries[name]) {
    return;
  }
  const response = await fetch('/libraries/' + name);
  this.libraries[name] = await response.text();
};
```

Reply:
Yep, we include a fetch polyfill in all our entry points here: https://github.com/code-dot-org/code-dot-org/blob/151f174856ceb4feef47af945cbde25214cb0219/apps/Gruntfile.js#L559